iceoryx_posh/include/iceoryx_posh/popo/base_subscriber.hpp

```diff
@@ -108,46 +108,38 @@ class BaseSubscriber
     ///
     void releaseQueuedSamples() noexcept;
 
-    template <uint64_t Capacity>
-    friend class WaitSet;
+  public:
+    /// @brief Only usable by the WaitSet, not for public use. Invalidates the internal triggerHandle.
+    /// @param[in] EventAccessor class constructable only by types which should have access to this method
+    /// @param[in] uniqueTriggerId the id of the corresponding trigger
+    /// @brief Only usable by the WaitSet, not for public use
+    void invalidateTrigger(const EventAccessor, const uint64_t trigger) noexcept;
+
+    /// @brief Only usable by the WaitSet, not for public use. Attaches the triggerHandle to the internal trigger.
+    /// @param[in] EventAccessor class constructable only by types which should have access to this method
+    /// @param[in] triggerHandle rvalue reference to the triggerHandle. This class takes the ownership of that handle.
+    /// @param[in] subscriberEvent the event which should be attached
+    void enableEvent(const EventAccessor,
```

Comment:
We ensuring that this method is present with the `EventAttachableConcept` at compile time. The client attorney pattern is a dirty pattern since we then would have to befriend here internal classes, therefore we pursued this approach.